### ISSUE TITLE
Moving cursor up after tab selection

### ISSUE THREAD

**Issue by chaoren** [junegunn/fzf#452]
Say you do `ctrl-t` and get this:
```
  some.other.garbage.foo
  foo.bar.baz.qux
  foo.bar.baz
  foo.bar
> foo
  5/5
> fo
```
Now to select `foo.bar`, `foo.bar.baz`, `foo.bar.baz.qux` you would need 3 `up`s to go up to `foo.bar.baz.qux`, 3 `tab`s to select all three, then `enter`. Or if you're a bit indecisive about which files to choose at the beginning: `up`, `tab`, `up`, `up`, `tab`, `up`, `up`, `tab`, `enter`.

Wouldn't it make more sense if it's just: `up`, `tab`, `tab`, `tab`, `enter`?
Basically, since the cursor starts at the bottom of the list, it's just more natural for it to go up as a default direction.

**Comment by chaoren** [comment]
Ah, I just noticed `ctrl`-`tab` does this. But I think this should be default, and `ctrl`-`tab` the other way around.

**Comment by junegunn** [comment]
Yeah, when I first implemented multi-selection [two years ago](https://github.com/junegunn/fzf/commit/ddf6e5ef1e55ae29cce1ab13c509581cb04c36a8) when fzf was still in Ruby, I thought it would be more conventional that `tab` brings your cursor down, and ever since then, I've kept it that way not to break backward-compatibility. And it's true that I'm almost always using `shift-tab`, unless `--reverse` is set. Maybe it's finally time for a change? Making `tab` move cursor away from the prompt, `shift-tab` the opposite. Is it better? Can it be confusing when we use `--reverse`? I wonder what others think.

Until I decide to change the default and release the next version, you can change the default with `--bind` option in `$FZF_DEFAULT_OPTS`.

**Comment by chaoren** [comment]
Yeah, I think more opinions would be nice.
I also think that `ctrl-tab`s should cancel out consecutive `tab`s. My hypothesis is that this would be a more useful purpose.

E.g., if we have this:
```
  some.other.garbage.foo
  foo.bar.baz.qux
  foo.bar.baz
  foo.bar
> foo
> fo
```
After `tab`, `tab`, `tab`, `ctrl-tab`, `ctrl-tab`, `ctrl-tab` it would become:
```
  some.other.garbage.foo
 >foo.bar.baz.qux
  foo.bar.baz
  foo.bar
>>foo
> fo
```
Again, more opinions would be nice.

**Comment by junegunn** [comment]
I'm going to add `toggle-in` and `toggle-out` actions in the next release. See the above commit for the details.

Your suggestion on cancelling out consecutive tabs is interesting and will be useful in that particular case. But I feel that it's going to be confusing in some other cases, so I'd like to just keep it simple.

**Comment by junegunn** [comment]
Closing due to the lack of community interest.

**Comment by rosshadden** [comment]
I bound this manually, as I found myself always hitting `<tab><ctrl-p><ctrl-p>` (because `<shift-tab>` doesn't seem to work in neovim for some reason).  I'm not sure if it should be default, but I am really glad you have the extensible binding system to facilitate such changes in user-land.
